smesh.go

```diff
@@ -0,0 +1,300 @@
+package ledger
+
+import (
+	"fmt"
+	//	"strconv"
+	//	"strings"
+)
+
+const CLA = 0x30
+const MAX_PACKET_LENGTH = 240
+
+const INS_GET_VERSION = 0x00
+const INS_GET_EXT_PUBLIC_KEY = 0x10
+const INS_GET_ADDRESS = 0x11
+const INS_SIGN_TX = 0x20
+
+const P1_UNUSED = 0x00
+const P1_RETURN = 0x01
+const P1_DISPLAY = 0x02
+const P1_HAS_HEADER = 0x01
+const P1_HAS_DATA = 0x02
+const P1_IS_LAST = 0x04
```

Comment:
I may be nitpicking here, but the way to make this look nice in Go is like this:

```suggestion
const (
	P1_UNUSED     = 0x00
	P1_RETURN     = 0x01
	P1_DISPLAY    = 0x02
	P1_HAS_HEADER = 0x01
	P1_HAS_DATA   = 0x02
	P1_IS_LAST    = 0x04
)
```

This can be applied in many places in this PR, I just picked this one as an example.

Reply:
Just to be clear, my version is after formatting.